src/python/pants/backend/helm/util_rules/manifest.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from enum import Enum, unique
+from pathlib import PurePath
+from typing import Any
+
+import yaml
+
+from pants.backend.helm.util_rules.yaml_utils import YamlElement, YamlPath
+from pants.engine.collection import Collection
+from pants.engine.engine_aware import EngineAwareParameter
+from pants.engine.fs import Digest, DigestContents, DigestSubset, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize, softwrap
+
+logger = logging.getLogger(__name__)
+
+
+@unique
+class StandardKind(Enum):
```

Comment:
Would it be better to use the [official Kubernetes Python client](https://github.com/kubernetes-client/python) to parse the YAML?

Reply:
thought about that although wasn't sure how welcoming we are to adding dependencies for such a very particular feature.

I'll spend some time with it looking at how it can be leveraged as a parser.

Reply:
Yeah, good point. We probably do not want the whole Kubernetes client as a dependency just for one backend.

@Eric-Arellano / @stuhood: Any advice here?

Reply:
For the Terraform backend, I recall that a third-party dependency was kept isolated to a venv pex invoked via a `Process` to avoid the dependency being a dependency of Pants itself. See https://github.com/pantsbuild/pants/blob/4210734ae6582d3a908f1ff67fe26ef62f0dd648/src/python/pants/backend/terraform/dependency_inference.py#L38-L51. Could that approach be used here?

Reply:
oh yeah, think I saw the same in the Docker backend for the Dockerfile parser which I used as inspiration for the Post Renderer implementation. Coud do something similar by spitting the info we care about into stdout and parsing it as JSON.

Reply:
removed the custom parsing of Kubernetes manifests but decided to use a different parser library as the official Kubernetes one, as that one wants to have a life connection to a cluster for some reason.

That library is: https://hikaru.readthedocs.io/en/latest/installation-quickstart.html and thanks to it, the code that parses the manifests has become way simpler.